packages/framework/src/hooks/__tests__/useRegisterBindings.test.ts

```diff
@@ -322,3 +341,194 @@ test("evaluates flutter style hex codes expressions", () => {
+});
+
+test("compare arrays with react fast compare isEqual", () => {
+  const newValues = {
+    allowedExtensions: ["jpg", "png", "pdf", "docs"],
+  };
+
+  const prevValues = {
+    values: {
+      allowedExtensions: ["jpg", "png", "pdf", "docs"],
+    },
+  };
+
+  expect(isEqual(newValues, prevValues.values)).toBe(true);
+});
+
+test("compare arrays with lodash isEqual", () => {
+  const newValues = {
+    allowedExtensions: ["jpg", "png", "pdf", "docs"],
+  };
+
+  const prevValues = {
+    values: {
+      allowedExtensions: ["jpg", "png", "pdf", "docs"],
+    },
+  };
+
+  expect(_.isEqual(newValues, prevValues.values)).toBe(true);
+});
+
+test("properly updates widgetState and returns correct values with forceState option", async () => {
+  // Render useState hook to manage values
+  const { result: countStat } = renderHook(() => useState(1));
+  const { result: fruitStat } = renderHook(() => useState("apple"));
+
+  const mockMethods = {
+    updateCount: countStat.current[1],
+    updateFruit: fruitStat.current[1],
+  };
+
+  // Initialize store with initial values
+  store.set(screenAtom, {
+    widgets: {},
+    data: {},
+    storage: {},
+  });
+
+  const { result: registerBindingResult, rerender } = renderHook(() =>
+    useRegisterBindings(
+      { count: countStat.current[0], fruit: fruitStat.current[0] },
+      "test",
+      mockMethods,
+    ),
+  );
+
+  // Initial value should be 1
+  expect(registerBindingResult.current.values).toMatchObject({
+    count: 1,
+    fruit: "apple",
+  });
+
+  // Update count using act to properly handle state updates
+  act(() => {
+    mockMethods.updateCount(3);
+    mockMethods.updateFruit("banana");
+  });
+
+  rerender();
+
+  await waitFor(() => {
+    expect(countStat.current[0]).toBe(3);
+    expect(fruitStat.current[0]).toBe("banana");
+    expect(registerBindingResult.current.values).toMatchObject({
+      count: 3,
+      fruit: "banana",
+    });
+
+    const finalScreen = store.get(screenAtom);
+    expect(finalScreen.widgets.test).toMatchObject({
+      values: {
+        count: 3,
+        fruit: "banana",
+      },
+      invokable: {
+        id: "test",
+        methods: mockMethods,
+      },
+    });
+  });
+});
+
+test("check registerBindings stale methods", async () => {
+  const fn1 = jest.fn();
+  const fn2 = jest.fn();
+  const testValues = { count: 1 };
+
+  // Initialize store with initial values
+  store.set(screenAtom, {
+    widgets: {},
+    data: {},
+    storage: {},
+  });
+
+  const { rerender } = renderHook(
+    (props) => useRegisterBindings(props.values, "test_stale", props.methods),
+    {
+      initialProps: {
+        values: testValues,
+        methods: { updateCount: fn1 },
+      },
+    },
+  );
+
+  // Rerender with new methods
+  rerender({
+    values: testValues,
+    methods: { updateCount: fn2 },
+  });
+
+  await waitFor(() => {
+    const finalScreen = store.get(screenAtom);
```

Comment:
Can you assert the resulting methods that get returned from `useRegisterBindings` is also updated?

Reply:
@evshi we are not getting resulting methods returned from `useRegisterBindings`, we only get `id`, `rootRef` and `values` from `useRegisterBindings`